### DISCUSSION TITLE
Are there any plans to support Qlora and P-Tuning v2

### DISCUSSION THREAD

**Discussion by GuWei007** [huggingface/peft#775] [General]
Are there any plans to support Qlora and P-Tuning v2

**Comment by ji-huazhong** [comment]
Qlora has been supported

**Comment by SumanthRH** [comment]
Qlora support was added in release v0.4.0, you can read up on how to make use of the integration here: https://huggingface.co/blog/4bit-transformers-bitsandbytes  

P-tuning v2 is already supported in the prefix-tuning implementation! P-tuning v2, as mentioned in the original paper, can be viewed as an "optimized and adapted implementation" of prefix tuning. In fact the implementation of prefix tuning in PEFT: https://github.com/huggingface/peft/blob/main/src/peft/tuners/prefix_tuning/model.py (as of v0.5.0) is almost the same as the implementation in P-tuning v2 (https://github.com/THUDM/P-tuning-v2/blob/main/model/prefix_encoder.py)

**Comment by clearsitedesigns** [comment]
It looks like this has been deprecated now as it's been stated. Does anyone know how to work around this?

prepare_model_for_kbittraining


This used to be the way to use it:

''' from peft import LoraConfig, get_peft_model

lora_alpha = 16
lora_dropout = 0.1
lora_r = 64

peft_config = LoraConfig(
    lora_alpha=lora_alpha,
    lora_dropout=lora_dropout,
    r=lora_r,
    bias="none",
    task_type="CAUSAL_LM"
)  '''

**Comment by clearsitedesigns** [comment]
IN case anyone else sees this it looks like the only change is:

```
from peft import LoraConfig, get_peft_model, TaskType

lora_alpha = 16
lora_dropout = 0.1
lora_r = 64

# Existing config
peft_config = LoraConfig(
    lora_alpha=lora_alpha,
    lora_dropout=lora_dropout,
    r=lora_r,
    bias="none",
    task_type="CAUSAL_LM"
)
```
